client/src/components/Form/Elements/FormSelect.vue

```diff
@@ -0,0 +1,182 @@
+<script setup lang="ts">
+import { computed, onMounted, watch, type ComputedRef } from "vue";
+import Multiselect from "vue-multiselect";
+
+type SelectValue = string | null;
+
+interface SelectOption {
+    label: string;
+    value: SelectValue;
+}
+
+const props = withDefaults(
+    defineProps<{
+        multiple?: boolean;
+        optional?: boolean;
+        options: Array<[string, string]>;
+        value?: Array<string> | string;
+    }>(),
+    {
+        multiple: false,
+        optional: false,
+        value: null,
+    }
+);
+
+const emit = defineEmits<{
+    (e: "input", value: SelectValue | Array<SelectValue>): void;
+}>();
+
+/**
+ * Determine dom wrapper class
+ */
+const cls: ComputedRef<string> = computed(() => {
+    return props.multiple ? "form-select-multiple" : "form-select-single";
+});
+
+/**
+ * Configure deselect label
+ */
+const deselectLabel: ComputedRef<string> = computed(() => {
+    return props.multiple ? "Click to remove" : "";
+});
+
+/**
+ * Translates input options for consumption by the
+ * select component into an array of objects
+ */
+const formattedOptions: ComputedRef<Array<SelectOption>> = computed(() => {
+    const result: Array<SelectOption> = props.options.map((option: [string, string]) => ({
+        label: option[0],
+        value: option[1],
+    }));
+    if (props.optional && !props.multiple) {
+        result.unshift({
+            label: "Nothing selected",
+            value: null,
+        });
+    }
+    return result;
+});
+
+/**
+ * Tracks if the select field has options
+ */
+const hasOptions: ComputedRef<Boolean> = computed(() => {
+    return formattedOptions.value.length > 0;
+});
+
+/**
+ * Configure selected label
+ */
+const selectedLabel: ComputedRef<string> = computed(() => {
+    return props.multiple ? "Selected" : "";
+});
+
+/**
+ * Tracks selected values
+ */
+const selectedValues: ComputedRef<Array<SelectValue>> = computed(() => {
+    return Array.isArray(props.value) ? props.value : [props.value];
+});
+
+/**
+ * Tracks current value and emits changes
+ */
+const currentValue = computed({
+    get: () => formattedOptions.value.filter((option: SelectOption) => selectedValues.value.includes(option.value)),
+    set: (val: Array<SelectOption> | SelectOption): void => {
+        if (Array.isArray(val)) {
+            if (val.length > 0) {
+                const values: Array<SelectValue> = val.map((v: SelectOption) => v.value);
+                emit("input", values);
+            } else {
+                emit("input", null);
+            }
+        } else {
+            emit("input", val.value);
+        }
+    },
+});
+
+/**
+ * Ensures that an initial value is selected for non-optional inputs
+ */
+function setInitialValue(): void {
+    if (props.value === null && !props.optional && hasOptions.value) {
```

Comment:
I'd be much more reassured about this if 'no value' was `undefined` ... `null` does seem like it should be a valid value and shouldn't be changed potentially without user interaction. The tests don't break so I guess this is fine for now.

Reply:
As I mentioned in other context, as of right now missing values are by default `null` for all input elements. In this particular case an `undefined` value passed as prop defaults to `null`.